### DISCUSSION TITLE
how does one configure email sending?

### DISCUSSION THREAD

**Discussion by jf** [jeremyevans/rodauth#407] [General]
I just tried to create a new account... and I get a ECONNREFUSED for localhost port 25:

![Screenshot 2024-04-09 at 10 05 30 PM](https://github.com/jeremyevans/rodauth/assets/7270/827445e2-fe23-46e1-b8cb-5bbe8aea2fac)

How do I configure the settings for the MTA that Rodauth will pass the email over to?

**Comment by janko** [comment]
You need to configure the [Mail](https://github.com/mikel/mail) gem with a valid delivery method.

```rb
Mail.defaults do
  delivery_method :smtp, ...
end
```

**Reply by jf** [reply]
got it. Thank you!

**Reply by jf** [reply]
one question about the way Mail is set up in rodauth_demo.rb: is there any reason why it's `::Mail.defaults` instead of `Mail.defaults` (at https://github.com/jeremyevans/rodauth/blob/747122ed504cc3732d5e14a60a352e9fc6ab9385/demo-site/rodauth_demo.rb#L28)?

If I do a test with `::Mail.__id__ == Mail.__id__`, it's `true`, so it's not as if there is another `Mail` in scope that could be called by mistake.

Or is this just to be explicit?

**Reply by jeremyevans** [reply]
It's to be explicit.  Using `Mail` makes it appear you are referring to `RodauthDemo::App::Mail` or `Roda::Mail` (and would break if either were defined separately).
